lake/branches/store.go

```diff
@@ -0,0 +1,106 @@
+package branches
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/brimdata/zed/lake/journal"
+	"github.com/brimdata/zed/pkg/storage"
+	"github.com/segmentio/ksuid"
+)
+
+var (
+	ErrExists   = errors.New("branch already exists")
+	ErrNotFound = errors.New("branch not found")
+)
+
+type Store struct {
+	store *journal.Store
+}
+
+func CreateStore(ctx context.Context, engine storage.Engine, path *storage.URI) (*Store, error) {
+	store, err := journal.CreateStore(ctx, engine, path, Config{})
+	if err != nil {
+		return nil, err
+	}
+	return &Store{store}, nil
+}
+
+func OpenStore(ctx context.Context, engine storage.Engine, path *storage.URI) (*Store, error) {
+	store, err := journal.OpenStore(ctx, engine, path, Config{})
+	if err != nil {
+		return nil, err
+	}
+	return &Store{store}, nil
+}
+
+func (s *Store) All(ctx context.Context) ([]Config, error) {
+	entries, err := s.store.All(ctx)
+	if err != nil {
+		return nil, err
+	}
+	list := make([]Config, 0, len(entries))
+	for _, entry := range entries {
+		branch, ok := entry.(*Config)
+		if !ok {
+			return nil, errors.New("corrupt branch config journal")
+		}
+		list = append(list, *branch)
+	}
+	return list, nil
+}
+
+func (s *Store) LookupByCommit(ctx context.Context, commit ksuid.KSUID) (*Config, error) {
+	list, err := s.All(ctx)
+	if err == nil {
+		for k, config := range list {
+			if config.Commit == commit {
+				return &list[k], nil
+			}
+		}
+	}
+	return nil, fmt.Errorf("%s: %w", commit, ErrNotFound)
+}
```

Comment:
This isn't in use, also it seems kind of weird to look up a branch by its commit, no?